Source Tiktok-Marketing: Add skan metrics to ads_reports_daily stream

katalin-schneider-knowunity opened the pull request:
## What

* Relates to  #48003


## How
Add below skan-specific metrics to ads_reports_daily stream (for tiktok_metrics: 

- skan_result
- skan_cost_per_result
- skan_conversion
- skan_cost_per_conversion
- skan_click_time_conversion
- skan_click_time_cost_per_conversion
- skan_click_time_conversion_rate
- skan_app_install
- skan_registration
- skan_complete_tutorial
- skan_purchase
- skan_cost_per_purchase
- skan_purchase_rate
- skan_total_purchase
- skan_cost_per_total_purchase
- skan_total_purchase_value
- skan_total_repetitive_active_pay_roas


## Review guide


## User Impact
* More columns for the users.

## Can this PR be safely reverted and rolled back?
<!--
* If unsure, leave it blank.
-->
- [x] YES 💚
- [ ] NO ❌

katalin-schneider-knowunity committed:
Source Tiktok-Marketing: Add skan metrics to ads_reports_daily stream

devwithkn commented:
@marcosmarxm @DanyloGL Hi, I saw that both of you worked on the previous adding of new columns. We are desperately waiting to get the skan columns to our data. Would be so lovely if you could consider adding them :) Best

DanyloGL committed:
Update metadata.yaml

DanyloGL committed:
Merge branch 'master' into add_scan_metrics_to_source_tiktok_marketing

DanyloGL commented:
/run-connector-tests
> **Connector CI Tests Started**
>
> These tests will leverage Airbyte's integration test credentials.
>
> [Check job output.](https://github.com/airbytehq/airbyte/actions/runs/19177149274)
> ✅ Connector CI Tests job completed successfully. See logs for details.

DanyloGL commented:
@devwithkn this PR is waiting for internal eng team review. Meantime, could you please take a look at unit tests [failure](https://github.com/airbytehq/airbyte/actions/runs/19177149274/job/54824810221)?

katalin-schneider-knowunity committed:
Update test_reports_hourly.py to include SKAN metrics and fix JSON formatting

- Add SKAN metrics to TestAdsReportHourly.metrics list in correct position
- Use json.dumps() for metrics and dimensions to match API request format
- Use json.dumps() for filtering parameter when include_deleted=True
- Fix query_params copying to ensure filtering is preserved for both date ranges

devwithkn commented:
Hi @marcosmarxm @DanyloGL: Is there any new update on when it will be implemented? Thank you!

darynaishchenko commented on airbyte-integrations/connectors/source-tiktok-marketing/manifest.yaml:
```diff
@@ -1518,7 +1518,24 @@ definitions:
+          "skan_total_purchase",
+          "skan_cost_per_total_purchase",
+          "skan_total_purchase_value",
+          "skan_total_repetitive_active_pay_roas",
```
have you tested this? 
just got an error `Invalid value for dimensions: hourly breakdown is not supported for skan metrics.` while running this stream locally
